src/background-view.cpp

```diff
@@ -0,0 +1,355 @@
+    struct wl_event_source *signal = nullptr;
+
+    public:
+    void init() override
+    {
+        grab_interface->name = transformer_name;
+        grab_interface->capabilities = 0;
+
+        option_changed = [=] ()
+        {
+            if (procs[output].view)
+            {
+                procs[output].view->close();
+                kill(procs[output].pid, SIGINT);
+                procs[output].view = nullptr;
+            }
+
+            procs[output].client = client_launch((std::string(command) + add_arg_if_not_empty(file)).c_str());
+        };
+
+        command.set_callback(option_changed);
+        file.set_callback(option_changed);
+
+        output->connect_signal("map-view", &view_mapped);
+
+        if (!signal)
+        {
+            signal = wl_event_loop_add_signal(wf::get_core().ev_loop, SIGCHLD, sigchld_handler, &procs);
+        }
+
+        procs[output].client = client_launch((std::string(command) + add_arg_if_not_empty(file)).c_str());
+        procs[output].view = nullptr;
+    }
+
+    wf::signal_connection_t view_mapped{[this] (wf::signal_data_t *data)
+    {
+        auto view = get_signaled_view(data);
+#if WLR_HAS_XWAYLAND
+        wlr_surface *wlr_surface = view->get_wlr_surface();
+        bool is_xwayland_surface = wlr_surface_is_xwayland_surface(wlr_surface);
+#endif
+        for (auto& o : wf::get_core().output_layout->get_outputs())
+        {
+            if (procs[o].client == view->get_client()
+#if WLR_HAS_XWAYLAND
+                || (is_xwayland_surface &&
+                /* For this match to work, the client must set _NET_WM_PID */
+                procs[o].pid == wlr_xwayland_surface_from_wlr_surface(wlr_surface)->pid)
+#endif
+                )
+            {
+#if WLR_HAS_XWAYLAND
+                if (is_xwayland_surface)
+                    view->set_decoration(nullptr);
+#endif
+                /* Move to the respective output */
+                wf::get_core().move_view_to_output(view, o);
+
+                /* A client should be used that can be resized to any size
+                 * If we set it fullscreen, the screensaver would be inhibited
+                 * so we send a resize request that is the size of the output
+                 */
+                view->set_geometry(o->get_relative_geometry());
+
+                /* Set it as the background */
+                o->workspace->add_view(view, wf::LAYER_BACKGROUND);
+
+                /* Make it show on all workspaces */
+                view->role = wf::VIEW_ROLE_DESKTOP_ENVIRONMENT;
+
+                procs[o].view = view;
+
+                break;
+            }
+        }
+    }};
+
+    std::string add_arg_if_not_empty(std::string in)
+    {
+        if (!in.empty())
+        {
+            return " \"" + in + "\"";
+        }
+        else
+        {
+            return in;
+        }
+    }
+
+    /* The following functions borrowed from weston */
+    int os_fd_set_cloexec(int fd)
```

Comment:
I'd like to see these functions outside of the class, preferably even in their own file which you can `#include` (so that it is clear which part is plugin, and which part is "boilerplate functions")